pulsar-protocols/grpc/src/main/proto/PulsarApi.proto

```diff
@@ -0,0 +1,888 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+syntax = "proto2";
```

Comment:
Could you please clarify why proto2 and not proto3 and maybe add it to [PIP-59](https://github.com/apache/pulsar/wiki/PIP-59%3A-gPRC-Protocol-Handler)?

Reply:
Because I don't like proto3 :smile:  : lack of optionals/nullables and default values...
Also the API is a derivation from the Pulsar binary protocol which is in proto2.
We could consider upgrading to proto3 later.